internal/storage/storage.go

```diff
@@ -0,0 +1,24 @@
+package storage
+
+import (
+	"errors"
+	"mmskazak/shorturl/internal/config"
+	"mmskazak/shorturl/internal/storage/infile"
+	"mmskazak/shorturl/internal/storage/inmemory"
+)
+
+type Storage interface {
+	GetShortURL(id string) (string, error)
+	SetShortURL(id string, targetURL string) error
+}
+
+func NewStorage(cfg *config.Config) (Storage, error) {
+	switch {
+	case cfg.FileStoragePath == "":
+		return inmemory.NewInMemory() //nolint:wrapcheck //ошибка обрабатывается далее
```

Comment:
Здесь к ошибке можно добавить контекст:


```suggestion
		s, err := inmemory.NewInMemory()
        if err != nil {
            return nil, fmt.Errorf("failed to initialize an in-memory store: %w", err)
        }
```

Reply:
```
	sm, err := inmemory.NewInMemory()
		if err != nil {
			return nil, fmt.Errorf("failed to initialize an in-memory store: %w", err)
		}
        return sm, nil
```